src/main/java/org/opensearch/security/filter/SecurityFilter.java

```diff
@@ -348,9 +348,13 @@ public void onFailure(Exception e) {
                 }
             } else {
                 auditLog.logMissingPrivileges(action, request, task);
-                String err = (injectedRoles == null) ?
-                        String.format("no permissions for %s and %s", pres.getMissingPrivileges(), user) :
-                        String.format("no permissions for %s and associated roles %s", pres.getMissingPrivileges(), injectedRoles);
+                String err = String.format("no permissions for %s and %s", pres.getMissingPrivileges(), user);
+                if(!pres.getMissingSecurityRoles().isEmpty()) {
+                    err = String.format("no permissions for %s to assume roles %s", user, pres.getMissingSecurityRoles());
```

Comment:
Update error message to remove assume role context.

Reply:
Done